Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py

```diff
@@ -0,0 +1,599 @@
+        return "PowderReduceP2D"
+
+    def seeAlso(self):
+        return ["SaveP2D"]
+
+    def PyInit(self):
+        # Input files
+        self.declareProperty(FileProperty('SampleData', '', action=FileAction.OptionalLoad, direction=Direction.Input),
+                             doc='Datafile that should be used.')
+        '''
+        self.declareProperty(WorkspaceProperty('SampleWorkspace', '',
+                                          direction = Direction.Input),
+                             doc='Workspace containing sample data.')
+        '''
+        self.declareProperty('DoIntensityCorrection',
+                             False,
+                             direction=Direction.Input,
+                             doc='If set to True you have to declare a vanadium measurement for intensity correction.')
+        self.declareProperty(FileProperty('VanaData', '', action=FileAction.OptionalLoad, direction=Direction.Input),
+                             doc='Vanadium measurement for intensity correction.')
+        '''
+        self.declareProperty(WorkspaceProperty('VanaWorkspace', '',
+                                          direction = Direction.Input),
+                             doc='Workspace containing vanadium data.')
+        '''
+        self.declareProperty('DoBackgroundCorrection',
+                             False,
+                             direction=Direction.Input,
+                             doc='If set to True you have to declare an empty can measurement for background correction.')
+        self.declareProperty(FileProperty('EmptyData', '', action=FileAction.OptionalLoad, direction=Direction.Input),
+                             doc='Empty measurement of the can for background correction.')
+        '''
+        self.declareProperty(WorkspaceProperty('EmptyWorkspace', '',
+                                          direction = Direction.Input),
+                             doc='Workspace containing empty data.')
+        '''
+        self.declareProperty(FileProperty('CalFile', '', action=FileAction.OptionalLoad, direction=Direction.Input),
+                             doc='Calibration file.')
+        self.declareProperty('DoEdgebinning', False, direction=Direction.Input, doc='If set to True you have to declare a BinEdges file.')
+        self.declareProperty(FileProperty('BinEdgesFile', '', action=FileAction.OptionalLoad, direction=Direction.Input),
+                             doc='BinEdges file used for edgebinning.')
+        grp1 = 'Input and Output Files'
+        self.setPropertyGroup('SampleData', grp1)
+        #self.setPropertyGroup('SampleWorkspace', grp1)
+        self.setPropertyGroup('DoIntensityCorrection', grp1)
+        self.setPropertyGroup('VanaData', grp1)
+        #self.setPropertyGroup('VanaWorkspace', grp1)
+        self.setPropertyGroup('DoBackgroundCorrection', grp1)
+        self.setPropertyGroup('EmptyData', grp1)
+        #self.setPropertyGroup('EmptyWorkspace', grp1)
```

Comment:
I have cleaned up the code and removed most commented out code. I have only left the commented out code for the input parameters of the algorithm AlignDetectors. I still want to get that part working at a later time, but I had some problems getting it working the way I wanted it to. I usually use calibration files for the algorithm, so its working fine for me, but at some time people might want to use a workspace instead of a calibration file for calibration.